### ISSUE TITLE
Cloning git repo on bitbucket and oauth2

### ISSUE THREAD

**Issue by vaclavvanik** [composer/composer#5160]
Hi,

I have problem with cloning git repo on bitbucket. I have set consumer key and secret in auth.json:

```
{
    "http-basic": {},
    "github-oauth": {},
    "gitlab-oauth": {},
    "bitbucket-oauth": {
        "bitbucket.org": {
            "consumer-key": "...",
            "consumer-secret": "..."
        }
    }
}
```

and I got

```
  - Installing vendor/repo (dev-master e408024)
    Cloning e408024dac5098371ae75f1de4acd6005a65a1bd
Executing command (CWD): git clone --no-checkout 'https://bitbucket.org/vendor/repo.git' '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo' && cd '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo' && git remote add composer 'https://bitbucket.org/vendor/repo.git' && git fetch composer
Downloading https://bitbucket.org/site/oauth2/access_token
Executing command (CWD): git clone --no-checkout 'https://x-token-auth:****@bitbucket.org/vendor/repo.git.git' '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo' && cd '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo' && git remote add composer 'https://x-token-auth:****@bitbucket.org/vendor/repo.git.git' && git fetch composer
Executing command (CWD): git --version
Failed: [RuntimeException] Failed to execute git clone --no-checkout 'https://x-token-auth:***@bitbucket.org/vendor/repo.git.git' '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo' && cd '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo' && git remote add composer 'https://x-token-auth:***@bitbucket.org/vendor/repo.git.git' && git fetch composer

Cloning into '/home/venca/PhpstormProjects/my-project/vendor/vendor/repo'...
remote: Not Found
fatal: repository 'https://x-token-auth:<token>@bitbucket.org/vendor/repo.git.git/' not found
```
It seems repo address has doubled .git extension. Is it bug or I have bad configuration anywhere?

I am using satis for handling private repos. Satis.json looks:

```
{
    "name": "my name",
    "homepage": "https://example.com",
    "repositories": [
        {"type": "vcs", "url": "https://bitbucket.org/vendor/repo.git"}
    ],
    "require-all": true,
    "archive": {
        "directory": "composer",
        "format": "zip",
        "skip-dev": false
    }
}

```

**Comment by alcohol** [comment]
I remember seeing this once before. Can you see if updating your Satis dependencies (in particular, `composer/composer`) resolves it?

**Comment by vaclavvanik** [comment]
Oh, I described my problem wrong. Problem is not in Satis. This happens when I run `composer update` in my project. I have private packages are in satis. Definitons is imho correct:

```
{
    "packages": {
        "vendor/repo": {
            "dev-master": {
                "name": "vendor/repo",
                "version": "dev-master",
                "version_normalized": "9999999-dev",
                "source": {
                    "type": "git",
                    "url": "https://bitbucket.org/vendor/repo.git",
                    "reference": "..."
                },
                ...
        },
```

**Comment by alcohol** [comment]
Having a hard time trying to reproduce this;

```
{
    "repositories": [{
        "type": "vcs",
        "url": "https://bitbucket.org/rbast/composer-bitbucket.git"
    }],
    "require": {
        "alcohol/composer-bitbucket": "dev-master"
    }
}
```

With my consumer key and secret provided in `auth.json`, I get the following error when I run install:

```
$ composer install -vvv
Reading ./composer.json
Loading config file /Users/rob/.composer/config.json
Loading config file /Users/rob/.composer/auth.json
Loading config file ./composer.json
Checking CA file /private/var/folders/3l/z23w626d1gl7xdz5t59f_3gh0000gn/T/composer-cacert-39eaf16e26c08cc90d728865f1c60406ee5d864d5963f72ede2a1cbb127f95ca.pem
Executing command (/private/tmp/testing-bitbucket): git branch --no-color --no-abbrev -v
Executing command (/private/tmp/testing-bitbucket): git describe --exact-match --tags
Reading /Users/rob/.composer/composer.json
Loading config file /Users/rob/.composer/config.json
Loading config file /Users/rob/.composer/auth.json
Loading config file /Users/rob/.composer/composer.json
Loading config file /Users/rob/.composer/auth.json
Reading /Users/rob/.composer/vendor/composer/installed.json
Loading composer repositories with package information
Downloading https://api.bitbucket.org/1.0/repositories/rbast/composer-bitbucket
Downloading https://api.bitbucket.org/1.0/repositories/rbast/composer-bitbucket


  [Composer\Downloader\TransportException]
  Invalid credentials for 'https://api.bitbucket.org/1.0/repositories/rbast/composer-bitbucket', aborting.


Exception trace:
 () at phar:///usr/local/bin/composer/src/Composer/Util/RemoteFilesystem.php:581
 Composer\Util\RemoteFilesystem->promptAuthAndRetry() at phar:///usr/local/bin/composer/src/Composer/Util/RemoteFilesystem.php:507
 Composer\Util\RemoteFilesystem->callbackGet() at n/a:n/a
 file_get_contents() at phar:///usr/local/bin/composer/src/Composer/Util/RemoteFilesystem.php:278
 Composer\Util\RemoteFilesystem->get() at phar:///usr/local/bin/composer/src/Composer/Util/RemoteFilesystem.php:99
 Composer\Util\RemoteFilesystem->getContents() at phar:///usr/local/bin/composer/src/Composer/Repository/Vcs/VcsDriver.php:107
 Composer\Repository\Vcs\VcsDriver->getContents() at phar:///usr/local/bin/composer/src/Composer/Repository/Vcs/GitBitbucketDriver.php:52
 Composer\Repository\Vcs\GitBitbucketDriver->getRootIdentifier() at phar:///usr/local/bin/composer/src/Composer/Repository/VcsRepository.php:136
 Composer\Repository\VcsRepository->initialize() at phar:///usr/local/bin/composer/src/Composer/Repository/ArrayRepository.php:179
 Composer\Repository\ArrayRepository->getPackages() at phar:///usr/local/bin/composer/src/Composer/DependencyResolver/Pool.php:104
 Composer\DependencyResolver\Pool->addRepository() at phar:///usr/local/bin/composer/src/Composer/Installer.php:402
 Composer\Installer->doInstall() at phar:///usr/local/bin/composer/src/Composer/Installer.php:228
 Composer\Installer->run() at phar:///usr/local/bin/composer/src/Composer/Command/InstallCommand.php:134
 Composer\Command\InstallCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:259
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:844
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:192
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:181
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:123
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:99
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:43
 require() at /usr/local/bin/composer:25
```

As I also stated in another thread/issue related to Bitbucket, there are very few people using it, and even fewer contributing code related to it. So it is not very well tested/proven nor is there a lot of knowledge present in our user-base. :-/

**Comment by alcohol** [comment]
See related also, #5166.

**Comment by Seldaek** [comment]
Should be fixed by #5272
